codegen/src/main/ops/org/nd4j/codegen/ops/SDBaseOps.kt

```diff
@@ -0,0 +1,1806 @@
+        Input(NUMERIC, "x") {count = AtLeast(1); description = "Input 1" }
+        Input(NUMERIC, "y") {count = AtLeast(1); description = "Input 2" }
+        Output(NUMERIC, "output"){ description = "Output SDVariable with values 0 and 1 based on where the condition is satisfied" }
+
+        Doc(Language.ANY, DocScope.ALL){
+            """ 
+                Less than operation: elementwise x < y
+                If x and y arrays have equal shape, the output shape is the same as these inputs.
+                Note: supports broadcasting if x and y have different shapes and are broadcastable.
+                Returns an array with values 1 where condition is satisfied, or value 0 otherwise.
+            """.trimIndent()
+        }
+    }
+
+    Op("lte") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.scalar.comparison"
+        javaOpClass = "ScalarLessThanOrEqual"
+        legacy = true
+        Input(NUMERIC, "x") { description = "Input array" }
+        Arg(NUMERIC, "y") { description = "Double value argument to use in operation" }
+        Output(NUMERIC, "output"){ description = "SDVariable with values 0 and 1 based on where the condition is satisfied" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Less than or equals operation: elementwise x <= y
+                Returns an array with the same shape/size as the input, with values 1 where condition is satisfied, or
+                value 0 otherwise
+            """.trimIndent()
+        }
+    }
+
+    Op("lte") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.custom"
+        javaOpClass = "LessThanOrEqual"
+        Input(NUMERIC, "x") { count = AtLeast(1); description = "Input 1" }
+        Input(NUMERIC, "y") { count = AtLeast(1);description = "Input 2" }
+        Output(NUMERIC, "output"){ description = "Output SDVariable with values 0 and 1 based on where the condition is satisfied" }
+        Doc(Language.ANY, DocScope.ALL){
+            """ 
+                Less than or equal to operation: elementwise x <= y
+                If x and y arrays have equal shape, the output shape is the same as these inputs.
+                Note: supports broadcasting if x and y have different shapes and are broadcastable.
+                Returns an array with values 1 where condition is satisfied, or value 0 otherwise.
+            """.trimIndent()
+        }
+    }
+
+    Op("matchCondition") {
+        javaPackage = "org.nd4j.linalg.api.ops.impl.transforms.bool"
+        javaOpClass = "MatchConditionTransform"
+        legacy = true
+        Input(NUMERIC, "in") { description = "Input" }
+        Arg(CONDITION, "condition") { description = "Condition" }
+        Output(NUMERIC, "output"){ description = "Boolean mask" }
+        Doc(Language.ANY, DocScope.ALL){
+            """
+                Returns a boolean mask of equal shape to the input, where the condition is satisfied - value 1 where satisfied, 0 otherwise
```

Comment:
As per lt, gt, eq etc - it's boolean out with true/false, not numerical with 1/0.
Old doc.

Reply:
Fixed.